content/en/platform/corda/5.0/developing-applications/getting-started/utxo-advanced-ledger-extensions/commands-example.md

```diff
@@ -0,0 +1,243 @@
+---
+menu:
+  corda5:
+    parent: "utxo-advanced-ledger-extensions"
+    identifier: corda5-utxo-ledger-extensions-examples
+    weight: 8050
+section_menu: corda5
+---
+
+# Building Basic Contract Design
+
+The following contract defines three commands; Create, Update and Delete. The `verify` function delegates these command types to `verifyCreate`, `verifyUpdate` and `verifyDelete` functions respectively, for example:
+
+```kotlin
+    public final class ExampleContract implements Contract {
+    private interface ExampleContractCommand extends Command { }
+  
+    public static class Create implements ExampleContractCommand { }
+    public static class Update implements ExampleContractCommand { }
+    public static class Delete implements ExampleContractCommand { }
+    
+    @Override
+    public void verify(UtxoLedgerTransaction transaction) {
+        
+        List<? extends ExampleContractCommand> commands = transaction
+                .getCommands(ExampleContractCommand.class);
+        
+        for (ExampleContractCommand command : commands) {
+            if (command instanceof Create) verifyCreate(transaction);
+            else if (command instanceof Update) verifyUpdate(transaction);
+            else if (command instanceof Delete) verifyDelete(transaction);
+            else throw new IllegalStateException("Unrecognised command type.");
+        }
+    }
+    
+    private void verifyCreate(UtxoLedgerTransaction transaction) {
+        // Verify Create constraints
+    }
+    
+    private void verifyUpdate(UtxoLedgerTransaction transaction) {
+        // Verify Update constraints
+    }
+    
+    private void verifyDelete(UtxoLedgerTransaction transaction) {
+        // Verify Delete constraints
+    }
+}
+  ```
```

Comment:
```suggestion
```kotlin
public final class ExampleContract implements Contract {
    
    private interface ExampleContractCommand extends Command { }
  
    public static class Create implements ExampleContractCommand { }
    public static class Update implements ExampleContractCommand { }
    public static class Delete implements ExampleContractCommand { }
    
    @Override
    public void verify(UtxoLedgerTransaction transaction) {
        List<? extends ExampleContractCommand> commands = transaction
                .getCommands(ExampleContractCommand.class);
        
        for (ExampleContractCommand command : commands) {
            if (command instanceof Create) verifyCreate(transaction);
            else if (command instanceof Update) verifyUpdate(transaction);
            else if (command instanceof Delete) verifyDelete(transaction);
            else throw new IllegalStateException("Unrecognised command type.");
        }
    }
    
    private void verifyCreate(UtxoLedgerTransaction transaction) {
        // Verify Create constraints
    }
    
    private void verifyUpdate(UtxoLedgerTransaction transaction) {
        // Verify Update constraints
    }
    
    private void verifyDelete(UtxoLedgerTransaction transaction) {
        // Verify Delete constraints
    }
}
```
```